airsonic-main/src/main/java/org/airsonic/player/controller/SubsonicRESTController.java

```diff
@@ -2145,7 +2148,15 @@ public void deleteUser(HttpServletRequest request, HttpServletResponse response)
         }
 
         String username = getRequiredStringParameter(request, "username");
-        if (org.airsonic.player.domain.User.USERNAME_ADMIN.equals(username)) {
+        org.airsonic.player.domain.User u = securityService.getUserByName(username);
+
+        if (u == null) {
+            error(request, response, ErrorCode.NOT_FOUND, "No such user: " + username);
+            return;
+        } else if (user.getUsername().equals(username)) {
+            error(request, response, ErrorCode.NOT_AUTHORIZED, "Not allowed to delete own user");
```

Comment:
What about "You're not allowed to delete your own user" instead?

Reply:
Since these messages are in the REST API, I kept them very terse, as no one should ever see them except developers.